crates/compiler/can/src/suffixed.rs

```diff
@@ -15,27 +15,13 @@ thread_local! {
     static SUFFIXED_ANSWER_COUNTER: Cell<usize> = const { Cell::new(0) };
 }
 
-/// Provide an intermediate answer expression and pattern when unwrapping a
-/// (sub) expression
-///
-/// e.g. `x = foo (bar!)` unwraps to `x = Task.await (bar) \#!a0 -> foo #!a0`
-fn next_suffixed_answer_pattern(arena: &Bump) -> (Expr, Pattern) {
-    // Use the thread-local counter
+/// Generates a unique identifier, useful for intermediate items during desugaring
```

Comment:
I wonder if we should also say something about why we are starting with a `#` here. The reason is that is it not possible for an ident to come from an application/platform author as it won't be parsable, it would be a comment. So these idents must be internal to the compiler. Also I used the `!` in there so it was easy to identify and I figured wouldn't conflict with other internal idents. I wonder if this is the best approach. I didn't look very hard to see if there were any other methods for getting these unique internal idents.

Reply:
yeah, I thought it's obvious 😁 
over time I think it would be more beneficial to use `next_suffixed_answer_pattern(name) -> "#suf_{counter}_{name}" to provide better debug info into desugared code